msteams-platform/m365-apps/specify-runtime-requirements.md

```diff
@@ -0,0 +1,135 @@
+
+- **Mutual dependencies:** When your app has specific app capabilities that must be loaded together to function properly, you can ensure your app is *only available* in Microsoft 365 hosts that support all capabilities.  
+
+- **Capability requirements:** When your app has capabilities with runtime requirements that aren't supported across all applicable Microsoft 365 hosts, you can ensure those capabilities are *seamlessly omitted* (rather than loaded but not functional) from your app experience running in those hosts.
+
+## Microsoft 365 host support
+
+The following Microsoft 365 host applications support the ability to specify runtime requirements in app manifest:
+
+|Microsoft 365 host application| Web | Desktop | Mobile |
+|---|---|---|---|
+| Teams| ✔️| ✔️| ✔️|
+| Outlook| ✔️| ✔️ (New Outlook only)| ✔️|
+| Microsoft 365 (Office)| | | |
+| Copilot for Microsoft 365 | | | |
+
+## Specify relationships between app capabilities (`elementRelationshipSet`)
+
+You can specify relationships among the individual capabilities of your app by including an `elementRelationshipSet` in your app manifest. Use this object to specify both [one-way dependencies](#one-way-dependencies) and [mutual dependencies](#mutual-dependencies) among app capabilities.
+
+The following app manifest capabilities can be specified as having one-way or mutual dependencies:
+
+- Tabs: personal (`staticTabs`) and configurable (`configurableTabs`)
+- Message extensions (`composeExtensions`), including individual commands
+- Bots (`bots`)
+
+Each app capability is defined by the newly introduced property, `id`, which maps to `botId` for bots, `entityId` for static tabs, and `id` for configurable tabs and message extensions. Teams and other Microsoft 365 hosts support apps that contain either a single configurable tab, a single message extension, or both. The ID property future-proofs your app in case host support expands to accommodate multiple instances of these capabilities.
+
+> [!IMPORTANT]
+> Ensure the relationships you create adhere to the following validation rules:
+>
+> 1. Elements specified under `elementRelationshipSet` must have definitions in app manifest. For example, an `element` or `commandId` listed in the `dependsOn` section of a `oneWayDependencies` object that doesn't have a corresponding definition in the manifest (with a matching `id` value) results in a manifest validation error. The `id` property must be specified in a `configurableTab` or a `composeExtension` for it to be recognized in an `elementRelationshipSet`.
+> 1. A given set of capabilities can only be grouped by a `mutualDependency` or `oneWayDependency`, but not both. For example, specifying both a one-way dependency (*A* depends on *B*) and a mutual dependency (*A* and *B* depend on each other) will result in a manifest validation error, because the *A depends on B* relationship is represented twice.
```

Comment:
```suggestion
> 1. A given set of capabilities can only be grouped by a `mutualDependency` or `oneWayDependency`, but not both. For example, specifying both a one-way dependency (*A* depends on *B*) and a mutual dependency (*A* and *B* depend on each other) results in a manifest validation error, because the *A depends on B* relationship is represented twice.
```